src/python.c

```diff
@@ -944,7 +944,7 @@ execute(ForeignScanState *node, ExplainState *es)
 		}
 		if (newqual != NULL)
 		{
-			PyObject   *python_qual = qualdefToPython((MulticornConstQual *) newqual, state->cinfos);
+			PyObject   *python_qual = qualdefToPython((MulticornConstQual *) newqual, state->qual_cinfos);
```

Comment:
It doesn't influence the existing behavior, since for non-agg queries `state->cinfos` is equal to `state->qual_cinfos`, given that [original table's tuple metadata](https://github.com/splitgraph/Multicorn/blob/agg-pushdown-v2-cu-1z461e4/src/multicorn.c#L2035) for non-agg queries matches the [target list tuple metadata](https://github.com/splitgraph/Multicorn/blob/agg-pushdown-v2-cu-1z461e4/src/multicorn.c#L658), which is not the case for aggregations. In short, I've added the `state->qual_cinfos` in order to track tuple metadata for the original table (i.e. columns of table in `FROM`) once the target tuple metadata (i.e. items in `SELECT` clause) deviates from it, which is the case in aggregations.

Note that regular `cinfos` are still being used for parsing the result of `execute` [here](https://github.com/splitgraph/Multicorn/blob/agg-pushdown-v2-cu-1z461e4/src/multicorn.c#L1117).